src/v/raft/recovery_stm.cc

```diff
@@ -380,7 +380,8 @@ bool recovery_stm::is_recovery_finished() {
      */
     return (is_up_to_date && quorum_writes) // fully caught up
            || _stop_requested               // stop requested
-           || _term != _ptr->term();        // leadership changed
+           || _term != _ptr->term()         // term changed
```

Comment:
why do we need term comparison if we have leadership check?

Reply:
The `is_leader` is transient condition. i.e. it might have been updated even if we are still the leader, we have to restart recovery in this case.

Reply:
but we still want to continue the recovery no? why do we want to stop with term? 

afaik, we should keep doing this thing until we are no longer a leader.

Reply:
If term changed that mean there was a leader, we have to stop recovery to account for the state changes, i.e. we should trigger it after followers state was updated on new leader.

Reply:
i see. you can be a leader in 2 different terms.

Reply:
just add a comment and LGTM on the 2 term thing.